### ISSUE TITLE
Can`t start scrapyd servise 

### ISSUE THREAD

**Issue by tonal** [scrapy/scrapyd#100]
I have scrapy-0.25 & scrapyd installed from repository http://archive.scrapy.org/ubuntu

scrapyd don't start from command line `start scrapyd`
In error log traceback:

```
$ cat /var/log/scrapyd/scrapyd.err 
Removing stale pidfile /var/run/scrapyd.pid
Unhandled Error
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/twisted/application/app.py", line 642, in run
    runApp(config)
  File "/usr/lib/python2.7/dist-packages/twisted/scripts/twistd.py", line 23, in runApp
    _SomeApplicationRunner(config).run()
  File "/usr/lib/python2.7/dist-packages/twisted/application/app.py", line 376, in run
    self.application = self.createOrGetApplication()
  File "/usr/lib/python2.7/dist-packages/twisted/application/app.py", line 441, in createOrGetApplication
    application = getApplication(self.config, passphrase)
--- <exception caught here> ---
  File "/usr/lib/python2.7/dist-packages/twisted/application/app.py", line 452, in getApplication
    application = service.loadApplication(filename, style, passphrase)
  File "/usr/lib/python2.7/dist-packages/twisted/application/service.py", line 405, in loadApplication
    application = sob.loadValueFromFile(filename, 'application', passphrase)
  File "/usr/lib/python2.7/dist-packages/twisted/persisted/sob.py", line 210, in loadValueFromFile
    exec fileObj in d, d
  File "/usr/lib/pymodules/python2.7/scrapyd/txapp.py", line 3, in <module>
    application = get_application()
  File "/usr/lib/pymodules/python2.7/scrapyd/__init__.py", line 14, in get_application
    return appfunc(config)
  File "/usr/lib/python2.7/dist-packages/promsoft/scrapyd/app.py", line 38, in application
    webservice = TCPServer(http_port, server.Site(Root(config, app)), interface=bind_address)
  File "/usr/lib/python2.7/dist-packages/promsoft/scrapyd/website.py", line 11, in __init__
    baseRoot.__init__(self, config, app)
  File "/usr/lib/pymodules/python2.7/scrapyd/website.py", line 27, in __init__
    servCls = load_object(servClsName)
  File "/usr/lib/pymodules/python2.7/scrapy/utils/misc.py", line 44, in load_object
    mod = import_module(module)
  File "/usr/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "/usr/lib/pymodules/python2.7/scrapyd/webservice.py", line 7, in <module>
    from scrapy.utils.txweb import JsonResource
exceptions.ImportError: No module named txweb

Failed to load application: No module named txweb
```

I see issue #60 closed.
May by update scrapyd paskage?

**Comment by vivekanand1101** [comment]
I'm getting exact same error. Scrapy version: 1.0.0 and scrapyd : 1.0.1. Any kind of help would be appreciated.

**Comment by adammfrank** [comment]
I'm having the same issue.  Ubuntu 14.04, Scrapy 1.0.0, scrapyd 1.0.1.

**Comment by Digenis** [comment]
Duplicates #60 and #96
The new release is not uploaded to pypi
and Scrapy>0.24 is not compatible with scrapyd<1.0.2

You can try installing the latest version from github for now
(just pass the url for cloning to easy_install or pip)

But the question that remains is,
are there any plans for a new release on pypi?

**Comment by adammfrank** [comment]
I installed both from pip and apt-get.  Both of those have the same problem, so it looks like the only way to install right now is from github.

**Comment by vivekanand1101** [comment]
@Digenis  Thanks for the help. I have installed the 0.24 version of Scrapy now, and it is working fine with Scrapyd 1.0.1

**Comment by adammfrank** [comment]
@Digenis Shouldn't the incompatibility be mentioned somewhere in either scrapy or scrapyd docs?

**Comment by tonal** [comment]
The packages have the latest scrapy (v0.25) and I can install it.
But working with him skrapid packages not.
It is inconvenient. IMHO.

**Comment by Digenis** [comment]
@tonal odd numbers are development/testing versions.
The latest stable, springing from 0.25, is 1.0.0

**Comment by tonal** [comment]
I install scrapy & scrapyd from http://archive.scrapy.org/ubuntu repository
latest scrapy version: 0.25.0-454-gfa1039f+1429829085
latest scrapyd version: 1.0~r111+1377637971

All versoin:
```
$ aptitude search scrapy
p   python-scrapy                                      - Python web scraping and crawling framework                   
p   python-scrapy-doc                                  - Python web scraping and crawling framework documentation     
v   python2.7-scrapy                                   -                                                              
p   scrapy                                             - Python web crawling and web scraping framework               
p   scrapy-0.14                                        - Python web crawling and scraping framework                   
p   scrapy-0.15                                        - Python web crawling and scraping framework                   
p   scrapy-0.16                                        - Python web crawling and scraping framework                   
p   scrapy-0.17                                        - Python web crawling and scraping framework                   
p   scrapy-0.18                                        - Python web crawling and scraping framework                   
p   scrapy-0.19                                        - Python web crawling and scraping framework                   
p   scrapy-0.20                                        - Python web crawling and scraping framework                   
p   scrapy-0.21                                        - Python web crawling and scraping framework                   
p   scrapy-0.22                                        - Python web crawling and scraping framework                   
p   scrapy-0.23                                        - Python web crawling and scraping framework                   
p   scrapy-0.24                                        - Python web crawling and scraping framework                   
i   scrapy-0.25                                        - Python web crawling and screen scraping framework            
c   scrapyd                                            - Scrapyd server                                               
p   scrapyd-0.14                                       - Scrapy Service                                               
p   scrapyd-0.15                                       - Scrapy Service                                               
p   scrapyd-0.16                                       - Scrapy Service                                               
p   scrapyd-0.17                                       - Scrapy Service                                               
p   scrapyd-deploy                                     - Scrapyd deployment tool                                      
p   scrapyorg                                          - scrapy.org website
```

**Comment by aguegu** [comment]
same issue happen to me with the  http://archive.scrapy.org/ubuntu repository.

scrapyd: 1.0~r111+1377637971
scrapy: 1.0.0.post2+g5303c66-1434981830

on
ubuntu 14.04

**Comment by tonypaek** [comment]
@vivekanand1101 How did you install the older version (0.24) of scrapy?

**Comment by adammfrank** [comment]
Install from here. http://scrapy.org/download/ 

When it downloads, execute "python setup.py install"

**Comment by aguegu** [comment]
pip install the package and manage it with supervisor can do the job. But I am looking for a more systematic way, :)

**Comment by jatinder85** [comment]
Latest versions of scrapy(1.0.3) and scrapyd(1.1.0) installed from pip seems to be compatible and working for me now.

**Comment by pablohoffman** [comment]
Is this fixed on scrapyd 1.1.0?. Can this issue be closed?

**Comment by boogheta** [comment]
It seems closed in 1.1 but shouldn't be closed until 1.1 is actually pushed into the apt repository, until then as explained in #114 you should at least update the documentation to handle it

**Comment by Digenis** [comment]
It is exactly as @boogheta says.
The [ubuntu repo](http://archive.scrapy.org/ubuntu/pool/main/s/scrapyd/) of scrapy.org still has version 1.0, from 2 years ago.
@nyov fixed it on 99ea9201dffe8891bcd95e3560c20d170fd5a33b, more than a year ago
but his fix didn't hit scrapinghub's ubuntu repo.
As a result, tickets #60, #100 and #114 are unresolved for ubuntu/debian users.

I don't think scrapyd 1.0 is intentionally left there.
@pablohoffman, is scrapy.org going to stop hosting scrapyd in its apt repo?
If yes, it needs to be removed from the docs **and** the apt repo to prevent users' confusion.
If not, someone should update it and add scrapyd 1.1 which fixes the issue.

**Comment by dangra** [comment]
I think we should remove it from docs or move the build process to Ubuntu PPAs or (travis-ci + bintray)

**Comment by dangra** [comment]
In seconds thoughts, not sure how many people is still using debian packages to distribute its apps but I'd like to vote to direct efforts on distributing a docker image of scrapyd instead of relying on debian packages. 

Within scrapinghub we are moving out of debian packages and more close to distribute docker images. Scrapinghub's apt repository is doomed and won't last for too long.

**Comment by Digenis** [comment]
Say, we use the scrapyd issue tracker as a sample.
There are many occurrences of people asking for ubunut/debian packages
and non so far for docker images.
I don't object to the removal of the apt repository.
I just want it to remain possible to install it in more traditional means.

**Comment by boogheta** [comment]
@Dangra If you want to drop the packages I can understand, but you could at the minimum just publish as the last version one that actually works.
It exists, and just needs to be pushed upstream, whereas currently the regular install just provides something unusable and with missing documentation on how to fix it.

**Comment by nyov** [comment]
I'll be sorry to see the debian packages go away. Is the current packaging build setup explained somewhere? I remember someone explaining a bit about that (dangra?), but not where.
I'd like to consider picking that up, if running and automating the build system isn't too involved.

**Comment by dangra** [comment]
The process to create the debian package is straightforward. After Installing the debian packaging helpers follow instructions in [Makefile.buildbot](https://github.com/scrapy/scrapyd/blob/master/Makefile.buildbot).

Right now the problem is that build is broken and I can't share details with you because buildbot is an internal Scrapinghub service. The build environment can't be that difficult to replicate using travis-ci, for those using debian packages most terminology must be familiar, and in the long term is better for everyone as the process becomes open and transparent.

There are at least 2 maintainers in ScrapyD whose are not related in any way to Scrapinghub, and I don't mind giving @nyov admin access if you (Nyov) want to restore the debian build system.

I suggest using travis-ci for building and bintray.com for hosting because I have been using both extensively with good results but whatever alternative you prefer is fine to me.

**Comment by Digenis** [comment]
closing due to lack of activity.
it will be solved in 1.2.1-post.1
